vtex-apps/store
# Fix/pageview

#### What is the purpose of this pull request?

Fix page view events firing in wrong situations. Fix structured data problems.

#### What problem is this solving?

events were fired at the wrong moment, with null products or wrong products.

#### How should this be manually tested?
https://fidtitle--alssports.myvtex.com/

In this workspace there is a console.log indicating when events are pushed to data layer, for debug.
navigate in this workspace and see that:
- Home page events fire just as expected
- when you enter a product-page it fires a console.log.
- Switching between product-pages (from a shelf.recommendations) and an event with the new product should fire
- Search page events are fine and fire after products are available.

#### Screenshots or example usage
![image](https://user-images.githubusercontent.com/4925068/62077073-9a104a80-b21f-11e9-9108-e235a8b32510.png)


#### Types of changes
